### PR TITLE
instantiate yumrepo & yum::config directly

### PR DESCRIPTION
**Pull request by igalic**
#### Pull Request (PR) description

Since we're not doing any dynamic creation of potentially unknown
resources, we want to use `yumrepo` and `yum::config` directly, rather
than going thru `Resource`

#### This Pull Request (PR) fixes the following issues

none; only my confusion when reading the code.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -133,8 +133,8 @@
 
     $normalized_repos.each |$yumrepo, $attributes| {
       if member($_managed_repos_minus_exclusions, $yumrepo) {
-        Resource['yumrepo'] {
-          $yumrepo: * => $attributes,
+        yumrepo { $yumrepo:
```
Comment (alexjfisher) [inline manifests/init.pp]: This isn't quite identical behaviour.  In the original form, I suspect there isn't a duplicate resource situation if the yumrepo has already been declared.  On the other hand, I don't think this is actually the case here looking through https://github.com/voxpupuli/puppet-yum/commit/1fd551cdcac27d6f1389b7ee3f86af6cfa664c3c

**Comment by alexjfisher** [comment]
@lamawithonel - let us know if we're accidentally breaking something! ;)

**Comment by lamawithonel** [comment]
I don't recall why I did that.  I worked on a replacement `yumrepo` resource type for a short time, but ultimately scrapped it.  It might have been a mix-up with another version to swap out the type.  In any case, you're right, it's unneeded the way it is now.

lgtm :+1:

**Comment by alexjfisher** [comment]
@lamawithonel Many thanks for the feedback.

**Pull request merged**